server/data_common/driver.py

```diff
@@ -0,0 +1,319 @@
+from abc import ABCMeta, abstractmethod
+from server_timing import Timing as ServerTiming
+import numpy as np
+import pandas as pd
+from server.data_common.fbs.matrix import encode_matrix_fbs
+from server.common.constants import Axis, DEFAULT_TOP_N
+from server.common.errors import FilterError, JSONEncodingValueError
+from server.compute.diffexp import diffexp_ttest
+from server.data_common.utils import jsonify_numpy
+from server.data_common.matrix_proxy import MatrixProxy
+from server.common.app_config import AppFeature, AppSingleConfig
+
+"""
+Sort order for methods
+1. Initialize
+2. Helper
+3. Filter
+4. Data & Metadata
+5. Computation
+"""
+
+
+class CXGDriver(metaclass=ABCMeta):
+
+    def __init__(self, config):
+        # config will normally be a type that inherits from AppConfig.
+        # the following is for backwards compatability with tests
+        if config is None:
+            config = AppSingleConfig()
+        elif type(config) == dict:
+            config = AppSingleConfig(**config)
+
+        # config is the application configuration
+        self.config = config
+
+        # parameters set by this data engine based on the data.
+        self.parameters = {}
+
+    def get_features(self):
+        features = {}
+        features["cluster"] = AppFeature("/cluster/")
```

Comment:
I would excise any dead code like this.  Also dead:
* /layout/var related
* "interactiveLimit" related

Reply:
Done